Lib/test/test_capi/test_watchers.py

```diff
@@ -294,6 +294,16 @@ class C2: pass
                 C2.hmm = "baz"
                 self.assert_events([C1, [C2]])
 
+    def test_all_watchers(self):
+        class C: pass
+        with ExitStack() as stack:
+            last_wid = -1
+            while last_wid < self.TYPE_MAX_WATCHERS - 1:
+                last_wid = stack.enter_context(self.watcher())
```

Comment:
maybe add a comment here explaining that we're doing it this way to get to the "last" watcher ID without making assumptions about pre-existing watches (other than assuming there are fewer than `TYPE_MAX_WATCHERS` pre-existing watchers).
I think another assumption is that watcher IDs are assigned based on the lowest unassigned ID? maybe it doesn't matter for this test, because you care about the high bit being set, not necessarily *all* bits.